platform/linux-generic/odp_pool.c

```diff
@@ -403,6 +430,12 @@ static odp_pool_t pool_create(const char *name, odp_pool_param_t *params,
 	block_size = ROUNDUP_CACHE_LINE(hdr_size + align + headroom +
 					data_size + tailroom);
 
+	/* Allocate extra memory for skipping packet buffers which cross page
+	 * boundaries. */
+	if (params->type == ODP_POOL_PACKET)
+		num_extra = 2 * (((uint64_t)(num * block_size) +
+				CONT_PAGE_SIZE - 1) / CONT_PAGE_SIZE);
```

Comment:
Why the `2 *` factor here? The number of block size "page" crossings is going to be limited to the rounded up number of pages:

```
<--------------------------> "n" pages in size
---+--------+--- ... ---+---- There can be at most n crossings,
                              (n - 1) if block happily starts on a page boundary.

```

Reply:
Also the extra buffers can cross page boundaries.

Reply:
The blocks are contiguous so you're computing how many of them can fit in a 2M huge page. Since blocks aren't themselves larger than a huge page I don't see what's wrong with my analysis.

Reply:
As you noted there can be at most N crossings with the original pool size ('num'). Now, to compensate for the skipped buffers the pool size is in practice increased by 'num_extra'. When the original pool size is large enough the extra buffers may again cross page boundary. 

Currently, a single buffer is 8KB, so a single 2MB page can hold max 256 buffers. With a pool size of 64K or larger the extra buffers will cross page boundary again.

Using 2x multiplication here is overkill. I'll update the patch and calculate the exact number of required extra buffers.